desc/compute/_stability.py

```diff
@@ -398,11 +398,27 @@ def _ideal_ballooning_gamma2(params, transforms, profiles, data, **kwargs):
 
     A_redo = B_inv @ A @ jnp.transpose(B_inv, axes=(0, 1, 3, 2))
 
-    w, _ = jnp.linalg.eigh(A_redo)
-    # max over "zeta" axis, still a function of rho, alpha, zeta0
-    gamma = jnp.real(jnp.max(w, axis=(2,)))
+    # TODO: Issue #1750
+    # Try jax.scipy.eigh_tridiagonal for increased performance
+    w, v = jnp.linalg.eigh(A_redo)
 
-    data["ideal ballooning lambda"] = gamma.flatten()
+    flat_w = w.flatten()
+
+    top_eigvals, top_flat_indices = jax.lax.top_k(flat_w, k=N_zeta - 2)
```

Comment:
can you summarize the difference here?

it used to take max over axis=2 which iirc is the "zeta" axis, so it gives one value per fieldline?

And now it gives N_zeta-2 values, independent of the number of fieldlines?

From the description in the PR/changelog I would think this topk stuff would be moved to the objective, and have this always return the full spectrum?

Reply:
Yeah, I see why this is confusing. I wanted k to be a fixed number > Neigvals so I chose N_zeta - 2. But the total number of eigenvalues should be (N_zeta-2) * N_alpha * N_zeta0. But, I don't want to store all the eigenvectors corresponding to these eigenvalues in memory.

For N_zeta ~ 2048, N_alpha = 8, N_zeta0 ~ 16, the matrix A and v occupy 4 GBs each. So someone running this on their CPU might end up getting OOMs even when they only needed the top eigenvalues.
I tried passing Neigvals as a static keyword argument but that fails during optimization. 

I don't know what the best solution is. For now, we can manually fix k to be a number << N_zeta and throw error of someone sets a large Neigvals in the BallooningStability objective.

Reply:
OK cool. My other question was whether we want `Neigvals` to be per-fieldline or global. I would think per fieldline, so that `Neigvals=1` is equivalent to the old behavior, and would likely also be better at preventing wack-a-mole type situations where you stabilize one fieldline but destabilize another.